hypershift-operator/controllers/hostedcluster/hostedcluster_controller_test.go

```diff
@@ -2241,34 +2305,737 @@ func TestComputeAWSEndpointServiceCondition(t *testing.T) {
+				},
+			},
+			expected: []*dto.MetricFamily{{
+				Name: pointer.String(hcmetrics.InitialRolloutDurationName),
+				Help: pointer.String("Time in seconds it took from initial cluster creation and rollout of initial version"),
+				Type: func() *dto.MetricType { v := dto.MetricType(1); return &v }(),
+				Metric: []*dto.Metric{{
+					Label: []*dto.LabelPair{
+						{
+							Name: pointer.String("name"), Value: pointer.String("hc"),
+						},
+						{
+							Name: pointer.String("namespace"), Value: pointer.String("any"),
+						},
+					},
+					Gauge: &dto.Gauge{Value: pointer.Float64(3600)},
+				}},
+			}},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			cluster := &hyperv1.HostedCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "hc",
+					Namespace:         "any",
+					CreationTimestamp: metav1.Time{Time: time.Time{}.Add(time.Hour)},
+				},
+				Status: hyperv1.HostedClusterStatus{
+					Version: &hyperv1.ClusterVersionStatus{
+						History: tc.updateHistory,
+					},
+				},
+			}
+
+			hcmetrics.HostedClusterInitialRolloutDuration.Reset()
+			reg := prometheus.NewPedanticRegistry()
+
+			// Capture metric.
+			if err := reg.Register(hcmetrics.HostedClusterInitialRolloutDuration); err != nil {
+				t.Fatalf("registering HostedClusterInitialRolloutDuration collector failed: %v", err)
+			}
+
+			// Report metric.
+			reportClusterVersionRolloutTime(cluster)
+
+			result, err := reg.Gather()
+			if err != nil {
+				t.Fatalf("gathering metrics failed: %v", err)
+			}
+
+			if diff := cmp.Diff(result, tc.expected); diff != "" {
+				t.Errorf("result differs from actual: %s", diff)
+			}
+		})
+	}
+}
+
+func TestReportLimitedSuportEnabled(t *testing.T) {
 	testCases := []struct {
 		name     string
-		hc       *hyperv1.HostedCluster
-		expected bool
+		labels   map[string]string
+		expected []*dto.MetricFamily
 	}{
 		{
-			name: "When it has HasBeenAvailable Annotation it should return true",
-			hc: &hyperv1.HostedCluster{
+			name:   "Limited support metric reported for current cluster",
+			labels: map[string]string{hyperv1.LimitedSupportLabel: "true"},
+			expected: []*dto.MetricFamily{{
+				Name: pointer.String(hcmetrics.LimitedSupportEnabledName),
+				Help: pointer.String("Indicates if limited support is enabled for each cluster"),
+				Type: func() *dto.MetricType { v := dto.MetricType(1); return &v }(),
+				Metric: []*dto.Metric{{
+					Label: []*dto.LabelPair{
+						{
+							Name: pointer.String("_id"), Value: pointer.String("fakeClusterID"),
+						},
+						{
+							Name: pointer.String("name"), Value: pointer.String("hc"),
+						},
+						{
+							Name: pointer.String("namespace"), Value: pointer.String("any"),
+						},
+					},
+					Gauge: &dto.Gauge{Value: pointer.Float64(1)},
+				}},
+			}},
+		},
+		{
+			name:   "Regular cluster, metric should not be reported for current cluster",
```

Comment:
All this test cases should be named following the convention When... it should... see TestHasBeenAvailable as an example.

This one is particularly confusing as the statement it's not true, it says "metric should not be reported for current cluster" and then the code validates that we report a metric with value 0.

Reply:
Makes sense, let me fix it.